packages/commonwealth/server/util/getCommentDepth.ts

```diff
@@ -1,29 +1,55 @@
+import { QueryTypes } from 'sequelize';
 import { DB } from 'server/models';
+import { CommentInstance } from '../models/comment';
 
 // getCommentDepth recursively calculates the depth of a comment,
 // then returns if the depth exceeds max, and the depth level
 export const getCommentDepth = async (
   models: DB,
-  comment,
+  comment: CommentInstance,
   maxIterations: number,
-  n?: number
 ): Promise<[exceeded: boolean, depth: number]> => {
-  if (!n) {
-    n = 0;
-  }
-
   if (!comment.parent_id) {
-    return [false, n];
+    return [false, 0];
   }
 
-  if (n >= maxIterations) {
-    return [true, maxIterations];
-  }
-  const parentComment = await models.Comment.findOne({
-    where: {
-      id: comment.parent_id,
-      chain: comment.chain,
-    },
-  });
-  return getCommentDepth(models, parentComment, maxIterations, n + 1);
+  const result: Array<{ comment_depth: number; max_depth_reached: boolean }> =
+    await models.sequelize.query(
+      `
+      WITH RECURSIVE CommentDepth AS (
+          SELECT id, parent_id, 0 AS depth, false AS max_depth_reached
+          FROM "Comments"
+          WHERE parent_id = :parentCommentId AND community_id = :communityId
+      
+          UNION ALL
+      
+          SELECT c.id, c.parent_id, cd.depth + 1,
+                 CASE
+                     WHEN cd.depth + 1 > :maxDepth THEN true
+                     ELSE false
+                     END AS max_depth_reached
+          FROM "Comments" c
+                   INNER JOIN CommentDepth cd ON c.id = cd.parent_id::INTEGER
+          WHERE cd.depth <= :maxDepth and community_id = :communityId
+      )
+      SELECT 
+          CASE 
+              WHEN bool_or(max_depth_reached) THEN :maxDepth
+              ELSE MAX(depth)
+          END AS comment_depth,
+          bool_or(max_depth_reached) AS max_depth_reached
+      FROM CommentDepth;
```

Comment:
Nice optimization.

Are there any current E2E/integration tests that verify the max depth case?

Reply:
There wasn't so I moved the existing unit tests which are technically useless now that the recursion is inside the query and refactored them as integration tests.

Reply:
Comment depth allows me to go to 10 (You can reproduce by always allowing client to reply to comments). Doesn't fail until trying to make comment of nested depth 11.

----

Out of scope of the PR, but is there any reason we need a max comment depth? Like we are just performing extra calculation + extra code to limit user flexibility, it doesn't seem like there are any PROs, only CONs (But maybe more of a product question).

Anyways maybe we could talk to product about it and refactor it sometime. Ideally we either:
1. Remove comment depth altogether
2. If it is really necessary, just add a field in the comment table that indicates its depth. Then we can just do a simple query on the field for the check.

Anyways aside from that, really good stuff. This query is really clever and taught me something new about sql.

Reply:
@kurtisassad from what I can tell this is exactly the same behaviour as what is on master right now i.e. 10 comments after disabling client-side block.

Reply:
Ah seems like you are correct. It does stop at 10. For some reason I had it go up to 11 when testing your branch, but maybe I had miscounted.